code/datums/progressbar.dm

```diff
@@ -57,16 +57,16 @@
 
 /datum/progressbar/Destroy()
 	if(last_progress != goal)
-		bar.icon_state = "[bar.icon_state]_fail"
-	for(var/I in user.progressbars[bar.loc])
+		bar?.icon_state = "[bar.icon_state]_fail"
+	for(var/I in user?.progressbars[bar.loc])
 		var/datum/progressbar/P = I
 		if(P != src && P.listindex > listindex)
 			P.shiftDown()
 
-	var/list/bars = user.progressbars[bar.loc]
-	bars.Remove(src)
-	if(!bars.len)
-		LAZYREMOVE(user.progressbars, bar.loc)
+	var/list/bars = user?.progressbars[bar?.loc]
+	bars?.Remove(src)
+	if(!bars?.len)
+		LAZYREMOVE(user?.progressbars, bar?.loc)
```

Comment:
The heck is going on here?

Reply:
Uhh those are way to many extra checks just for that gonna fix it

Reply:
nevermind thats a datum why did i even put this there then datums are not even getting called